docs/quick_tutorial/debugtoolbar.rst

```diff
@@ -65,7 +65,8 @@ Steps
         $VENV/bin/pserve development.ini --reload
 
 #.  Open http://localhost:6543/ in your browser.
-    See the handy toolbar on the right.
+    See the handy toolbar on the right. ``note: the Response object should contain <body> html or it won't be shown unless an error is introduced.``
```

Comment:
If you follow the tutorial correctly, does this issue actually occur? If not, then please remove this note.

Also as an FYI, the correct syntax for notes in reStructuredText is the following, with proper indentation:

```
.. note::

    This is my note.
    It is multi-line, one sentence per line.
```

or

```
.. note::

    This is my single sentence note on one line.
```

or

```
.. note:: This is my single sentence note on one line.
```

I prefer the first two formats to the last for all admonitions because it makes it easier to add more text, if needed.

Reply:
tutorial approach suggests **copying** the previous folder of the tutorial to the next steps of the tutorial. 

https://docs.pylonsproject.org/projects/pyramid/en/latest/quick_tutorial/tutorial_approach.html

so I made that change in previous steps of the tutorial ( which was a few days ago ). 

when I refreshed the page I didn't see the debugtoolbar and I already **fogot**  that I made this small change in the response string.( I am just discovering the framework so I don't want any html in my response. ) 

but you know beginners want to see something that works and build upon and that might prevent new comers from trying different things to get it to work given that they are new.

a suggested edit also will be introducing an error with `Xresponse` instead of `Response` as a step of the tutorial not as an extra credit.

Reply:
> tutorial approach suggests copying the previous folder of the tutorial to the next steps of the tutorial.

In the Quick Tutorial, it's a _requirement_ to copy the previous step for all but a few steps. You'll see why when the tutorial forks from a much earlier step.

> when I refreshed the page I didn't see the debugtoolbar and I already fogot that I made this small change in the response string.( I am just discovering the framework so I don't want any html in my response. )

If you experiment or make mistakes, then that's great for your learning. However we will not include experiments in the tutorial documentation. It is intended to be a "_Quick_ Tutorial". We also include the source code in the Pyramid repository, and we recommend that users clone it so that they can compare their code with official source code to find their mistakes.

FYI, the debugtoolbar is intended to be used only in development environments. The [Usage documentation](https://docs.pylonsproject.org/projects/pyramid_debugtoolbar/en/latest/#usage) states:

> For any HTML responses generated by the application, a link to the toolbar for the current page will be available in the upper right corner, provided the response contains a closing `</body>` tag.

We prefer to reference documentation with links to it instead of duplicating it.

> but you know beginners want to see something that works and build upon and that might prevent new comers from trying different things to get it to work given that they are new.

There is a future step, [14: AJAX Development With JSON Renderers](https://docs.pylonsproject.org/projects/pyramid/en/latest/quick_tutorial/json.html) that covers an example of how to return a JSON response with a content-type of `application/json` instead of the default content-type of `text/html`.

We have a [few more official tutorials](https://docs.pylonsproject.org/projects/pyramid/en/latest/index.html#tutorials), and there are some [community maintained tutorials](https://docs.pylonsproject.org/projects/pyramid_tutorials/en/latest/). There is also a very active project [pyramid_openapi3](https://github.com/pylons/pyramid_openapi3) that provides [several demo apps](https://github.com/pylons/pyramid_openapi3#demo--examples).

> a suggested edit also will be introducing an error with Xresponse instead of Response as a step of the tutorial not as an extra credit.

I'm not sure how that would fit into the Quick Tutorial. I'd have to see a draft of the proposal. It might be a better candidate for the [Pyramid Community Cookbook](https://docs.pylonsproject.org/projects/pyramid_cookbook/en/latest/).

There's also a great [community](https://trypyramid.com/community.html) as a resource for new developers. We were all new developers to Pyramid at one point, so we know how it is.